ui/core/worker_pool.ts

```diff
@@ -125,7 +125,7 @@ export class WorkerPool {
 	async raidSimAsync(request: RaidSimRequest, onProgress: WorkerProgressCallback): Promise<RaidSimResult> {
 		const worker = this.getLeastBusyWorker();
 		worker.log('Raid sim request: ' + RaidSimRequest.toJsonString(request));
-		const id = worker.makeTaskId();
+		const id = request.requestId || worker.makeTaskId();
```

Comment:
Could we please change `worker.makeTaskId()` to either use the `randomUUID` function, or re-use the one we do for requestIDs? Otherwise we wil now have 3 different generators

Reply:
I moved generateRequestId() to worker_pool.ts and replaced makeTaskId() with it. Also makes it easier to follow what's happening in logs.